lib/wget.js

```diff
@@ -82,18 +83,28 @@ function download(src, output, options, _parentEvent, redirects) {
                 download(newLocation, output, options, downloader, redirects);
             }
 
-            if (res.statusCode === 200) {
+            if (res.statusCode === 200 || res.statusCode === 206) {
                 downloadedSize = 0;
                 fileSize = Number(res.headers['content-length']);
 
                 // If content length header is not sent, there is no way to determine file size.
                 if (isNaN(fileSize) === true) {
                     fileSize = null;
                 }
-                writeStream = fs.createWriteStream(output, {
-                    flags: 'w+',
+
+                const flags = res.statusCode === 206 ? 'a' : 'w+';
+                const writeStreamOptions = {
+                    flags,
                     encoding: 'binary'
-                });
+                };
+
+                if (!isNaN(options.start)) {
+                    fs.closeSync(fs.openSync(output, 'a'));
+                    writeStreamOptions.start = options.start;
+                    writeStreamOptions.flags = 'r+';
```

Comment:
Are you sure this would work with r+?

```
'r+' - Open file for reading and writing. An exception occurs if the file does not exist.
```

Please try delete the file completely and download a new file?
I think we need to
1. create an empty file only if the file doesn't already exist (we should prob check)
2. subsequent we just need to open the file with 'w' or 'a'?

Reply:
Yeah, r+ works, r+ is the same as w without the truncation.
L:102 handles the creation of files if it doesn't exist.

Reply:
1. I meant if the file is already exist, we can skip creation/open file.
2. But we are opening the file for "writing", not "reading"? The system shouldn't allow you to write to the file since you didn't request for "write" operation (either a or w)

Reply:
1. Yeah, L102 does exactly that because of 'a', it checks if the file exists, then creates it if it doesn't.
2. 'r+' does have write access https://nodejs.org/api/fs.html#fs_file_system_flags

Reply:
1. But it also opens a file where we need to close, additional 2 IO operation (open & close) vs check.
2. I see I kept missing the "writing" part, so the difference is one truncates and another doesn't.

Reply:
Ah true, check added.